src/SSHDebugPS/Connection.cs

```diff
@@ -126,6 +127,19 @@ internal string GetUserHomeDirectory()
             return _remoteSystem.FileSystem.GetDirectory(liblinux.IO.SpecialDirectory.Home).FullPath;
         }
 
+        public string AttachToProcess(int pid, string preAttachCommand)
+        {
+            var gdbStart = new liblinux.Services.GdbServerStartInfo();
+            gdbStart.ProcessId = pid;   // indicates an attach operation
+            gdbStart.PreLaunchCommand = preAttachCommand;
+            _gdbserver = _remoteSystem.Services.GdbServer.Start(gdbStart);
```

Comment:
>Start [](start = 58, length = 5)

Do you know what the error reporting is like if gdbserver fails to attach?

Reply:
Start throws on failure with an explicit message.